python/lsst/daf/butler/registry/obscore/_records.py

```diff
@@ -394,12 +394,16 @@ def make_universe_records(self, ref: DatasetRef) -> Record:
         dataId = ref.dataId
         record: dict[str, str | int | float | UUID | None] = {}
 
-        instrument_name = cast(str, dataId.get("instrument"))
+        instrument_name = cast(str | None, dataId.get("instrument"))
```

Comment:
mypy starts getting annoyed if I try to use a default fallback instrument here because dataId.get doesn't allow a None default.

Reply:
I think `DataCoordinate.get` probably needs some `@typing.override` work; that's the usual pattern for `Mapping.get`-likes, and I don't think I realized that back when I first wrote `DataCoordinate`.

Reply:
Hmm, with `default: None` it should always return `None`? There is already this method:

    def get(self, key: str, default: DataIdValue | None = None) -> DataIdValue | None:

which I think should cover it.

Reply:
Except it doesn't cover it. Maybe mypy only looks at the overloads and not the implementation itself?

Reply:
Okay. If I explicitly add the `DataIdValue | None` option as an overload things do work. @andy-slac does this look okay to you?